header-rewrite-filter/header_processor.cc

```diff
@@ -221,46 +220,76 @@ namespace HeaderRewriteFilter {
+            if (status != absl::OkStatus()) {
+                return status;
+            }
+
+            // parse optional argument
+            std::string string_to_compare;
+            if (match_type != Utility::MatchType::Found) {
+                string_to_compare = std::string(*(start + 4));
+            }
 
             switch (match_type) {
                 case Utility::MatchType::Exact:
-                    source_ = std::string(*(start + 1));
-                    matcher_ = [string_to_compare](std::string source) { return source.compare(string_to_compare) == 0; };
+                    matcher_ = [string_to_compare](std::string source) { return (source.compare(string_to_compare) == 0); };
                     break;
                 case Utility::MatchType::Prefix:
-                    source_ = std::string(*(start + 1));
                     matcher_ = [string_to_compare](std::string source) { return string_to_compare.compare(0, source.size(), source, 0, source.size()) == 0; };
                     break;
                 case Utility::MatchType::Substr:
-                    source_ = std::string(*(start + 1));
                     matcher_ = [string_to_compare](std::string source) { return source.find(string_to_compare) != std::string::npos; };
                     break;
-                // TODO: implement this
-                case Utility::MatchType::Found:
+                case Utility::MatchType::Found: // urlp found or hdr found
+                    matcher_ = [](std::string source) { return true; };
                     break;
                 default:
                     return absl::InvalidArgumentError("invalid match type");
             }
 
-        } catch (const std::exception& e) {
+        } catch (const std::exception& e) { // could throw bad optional access if config syntax is wrong
             return absl::InvalidArgumentError("error parsing boolean expression -- " + std::string(e.what()));
         }
 
+
         return absl::OkStatus();
     }
 
-    // return status and bool result
-    std::tuple<absl::Status, bool> SetBoolProcessor::executeOperation(bool negate) {
-        try {
-            const bool result = matcher_(source_);
-            const bool apply_negation = negate ? !result : result;
-            return std::make_tuple(absl::OkStatus(), apply_negation);
-        } catch (std::exception& e) {
-            return std::make_tuple(absl::InvalidArgumentError("failed to perform boolean match -- " + std::string(e.what())), false);
+    std::tuple<absl::Status, bool> SetBoolProcessor::executeOperation(Http::RequestOrResponseHeaderMap& headers, bool negate) {
+        const auto arguments = StringUtil::splitToken(function_argument_, ",", false, true);
+        switch (function_type_) {
+            case Utility::FunctionType::GetHdr:
+            {
+                const auto result = Utility::getHeaderValue(headers, arguments, matcher_, negate);
```

Comment:
I don't think we need the matcher or negation passed down to `getHeaderValue`. Those concepts are related to the bool processor itself. E.g. in the future we might just want a generic set-var using a header value. I imagine something like

```
const bool result = matcher_(getHeaderValue(headers, arguments));
```